src/main/java/ca/corefacility/bioinformatics/irida/repositories/sequencefile/SequenceConcatenationRepository.java

```diff
@@ -1,11 +1,34 @@
 package ca.corefacility.bioinformatics.irida.repositories.sequencefile;
 
+import java.util.Set;
+
+import org.springframework.data.jpa.repository.Query;
+
 import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceConcatenation;
+import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequencingObject;
 import ca.corefacility.bioinformatics.irida.repositories.IridaJpaRepository;
 
 /**
  * Repository for {@link SequenceConcatenation}s
  */
 public interface SequenceConcatenationRepository extends IridaJpaRepository<SequenceConcatenation, Long> {
 
+	/**
+	 * Get the {@link SequenceConcatenation} for a given {@link SequencingObject}
+	 *
+	 * @param sequencingObject the {@link SequencingObject}
+	 * @return a {@link SequenceConcatenation}
+	 */
+	@Query("select sc from SequenceConcatenation sc where sc.concatenated = ?1")
+	public SequenceConcatenation findConcatenatedSequencingObject(SequencingObject sequencingObject);
+
+	/**
+	 * Get a set of {@link SequenceConcatenation}s if the given {@link SequencingObject} is a source
+	 *
+	 * @param sequencingObject the {@link SequencingObject}
+	 * @return the set of {@link SequenceConcatenation}s
+	 */
+	@Query("select sc from SequenceConcatenation sc where ?1 IN elements(sc.sources)")
+	public Set<SequenceConcatenation> findConcatenatedSequencingObjectSources(SequencingObject sequencingObject);
```

Comment:
**nitpick (blocking):** Could you update this method name to be `findConcatenatedSequencingObjectsBySource`, I was initially confused as to what this method did until I read the comments. From the original name I had assumed that given a `SequencingObject` I would find all of it's sources.

Reply:
Please see bbcf6a3ebd86b3d643093e481d0bd90ef16581f8.